kion/resource_aws_account.go

```diff
@@ -208,6 +208,90 @@ func resourceAwsAccount() *schema.Resource {
 	}
 }
 
+func createAwsAccount(ctx context.Context, c *hc.Client, postCacheData hc.AccountCacheNewAWSCreate, d *schema.ResourceData) (diag.Diagnostics, int) {
+	var diags diag.Diagnostics
+
+	// Lock to ensure one account creation process at a time as AWS Orgs cannot handle more than one account creation at a time.
+	awsAccountCreationMux.Lock()
+	defer awsAccountCreationMux.Unlock()
+
+	// Populate organizational unit details from Terraform resource data, if provided by user.
+	if err := populateOrgUnitFromResourceData(c, &postCacheData, d); err != nil {
+		return diag.FromErr(err), 0
+	}
+
+	// Send the POST request to create the AWS account.
+	respCache, err := c.POST("/v3/account-cache/create?account-type=aws", postCacheData)
+	if err != nil || respCache.RecordID == 0 {
+		if err == nil {
+			err = fmt.Errorf("received item ID of 0")
+		}
+		return diag.Errorf("Unable to create AWS Account: %v", err), 0
```

Comment:
I use to include the post body in a the detail message of the Diagnostic struct.  I found it helpful when debugging to understand what was sent to the API.  This omits that.  I guess it's ok to omit that as a regular user may not benefit from it, but just pointing it out in case omitting it wasn't intentional.